### ISSUE TITLE
Predicting emotions for unseen tweets

### ISSUE THREAD

**Issue by khairiyah98** [nikicc/twitter-emotion-recognition#15]
Hi, can I know if we can use these to predict emotions for any tweets? Or do we need to train them downstream?

**Comment by nikicc** [comment]
> Hi, can I know if we can use these to predict emotions for any tweets?

Yes, the models are already trained and can be used directly.

> Or do we need to train them downstream?

No need. They are already trained.

**Comment by khairiyah98** [comment]
Alright, thank you!

May I know what are the preprocessing steps required for the tweet input into the model? For e.g. Do I need to remove emoticons/emojis?

**Comment by nikicc** [comment]
@khairiyah98 I don't think you need to do any preprocessing. Feel free to leave the emoticons in.

**Comment by Akshar-code** [comment]
Hii @nikicc in your demo.ipynb file i can see you fed the tweets are a tuple
![Screenshot 2021-02-15 at 4 10 30 PM](https://user-images.githubusercontent.com/59618640/107936182-58352980-6fa8-11eb-8bf8-ec946d7bba91.png)

Im wondering how i can do this if i have a csv file with rows of tweets? please help me
![Screenshot 2021-02-15 at 4 11 40 PM](https://user-images.githubusercontent.com/59618640/107936305-7e5ac980-6fa8-11eb-9132-3681df881b22.png)

Also @khairiyah98 have you made any progress with this aspect?

**Comment by nikicc** [comment]
Hi @Akshar-code!

> Hii @nikicc in your demo.ipynb file i can see you fed the tweets are a tuple

It's actually a list, not a tuple. But tuple might also work, not sure 🤷‍♂️.

> Im wondering how i can do this if i have a csv file with rows of tweets? please help me

So if you have a file with the structure as shown in the above image, you just need to read it line by line into a list of strings. Check this [StackOverflow answers](https://stackoverflow.com/questions/3277503/how-to-read-a-file-line-by-line-into-a-list) for code snippets that do exactly this.
